exchanges/gateio/gateio_wrapper.go

```diff
@@ -837,103 +1657,262 @@ func (g *Gateio) GetOrderHistory(ctx context.Context, req *order.GetOrdersReques
-			Pair:                 pair,
+	case asset.Options:
+		for x := range req.Pairs {
+			fPair := req.Pairs[x].Format(format)
+			var optionOrders []OptionTradingHistory
+			optionOrders, err = g.GetOptionsPersonalTradingHistory(ctx, fPair.String(), fPair.Upper(), 0, 0, req.StartTime, req.EndTime)
+			if err != nil {
+				return nil, err
+			}
+			for o := range optionOrders {
+				detail := order.Detail{
+					OrderID:   strconv.FormatInt(optionOrders[o].OrderID, 10),
+					Amount:    optionOrders[o].Size,
+					Price:     optionOrders[o].Price,
+					Date:      optionOrders[o].CreateTime.Time(),
+					Exchange:  g.Name,
+					Pair:      fPair,
+					AssetType: req.AssetType,
+				}
+				detail.InferCostsAndTimes()
+				orders = append(orders, detail)
+			}
 		}
-		detail.InferCostsAndTimes()
-		orders[i] = detail
+	default:
+		return nil, fmt.Errorf("%s does not support %s", g.Name, req.AssetType)
 	}
 	return req.Filter(g.Name, orders), nil
 }
 
-// AuthenticateWebsocket sends an authentication message to the websocket
-func (g *Gateio) AuthenticateWebsocket(ctx context.Context) error {
-	return g.wsServerSignIn(ctx)
-}
-
-// ValidateCredentials validates current credentials used for wrapper
-// functionality
-func (g *Gateio) ValidateCredentials(ctx context.Context, assetType asset.Item) error {
-	_, err := g.UpdateAccountInfo(ctx, assetType)
-	return g.CheckTransientError(err)
-}
-
-// FormatExchangeKlineInterval returns Interval to exchange formatted string
-func (g *Gateio) FormatExchangeKlineInterval(in kline.Interval) string {
-	return strconv.FormatFloat(in.Duration().Seconds(), 'f', 0, 64)
-}
-
 // GetHistoricCandles returns candles between a time period for a set time interval
 func (g *Gateio) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
 	req, err := g.GetKlineRequest(pair, a, interval, start, end)
 	if err != nil {
 		return nil, err
 	}
-
-	timeSeries, err := g.GetSpotKline(ctx, KlinesRequestParams{
-		Symbol:   req.RequestFormatted.String(),
-		GroupSec: g.FormatExchangeKlineInterval(req.ExchangeInterval),
-		HourSize: int(time.Since(req.Start).Hours()),
-	})
-	if err != nil {
-		return nil, err
+	var listCandlesticks []kline.Candle
+	switch a {
+	case asset.Spot, asset.Margin, asset.CrossMargin:
+		var candles []Candlestick
+		candles, err = g.GetCandlesticks(ctx, req.RequestFormatted, 0, start, end, interval)
+		if err != nil {
+			return nil, err
+		}
+		listCandlesticks = make([]kline.Candle, len(candles))
+		for x := range candles {
+			listCandlesticks[x] = kline.Candle{
+				Time:   candles[x].Timestamp,
+				Open:   candles[x].OpenPrice,
+				High:   candles[x].HighestPrice,
+				Low:    candles[x].LowestPrice,
+				Close:  candles[x].ClosePrice,
+				Volume: candles[x].QuoteCcyVolume,
+			}
+		}
+	case asset.Futures, asset.DeliveryFutures:
+		if req.RequestFormatted.Quote.MatchAny(currency.USD, currency.USDT, currency.BTC) {
```

Comment:
This seems like a job for `getSettlementFromCurrency` given 
![image](https://user-images.githubusercontent.com/9261323/219277955-36a2d277-d6b1-4543-8ee1-e41074f07196.png)